### PR TITLE
Adding a sample inventory and basic instructions for following a BYO …

### PR DESCRIPTION
**Pull request by etsauer**
…infra model

#### What does this PR do?
@oybed @logandonley as discussed last week, this is my first attempt to make the host prep/byo infra story a thing.

FYI, I am targeting the use of this as training material for our practice architect enablement session next week.

#### How should this be manually tested?
Read through the README, and follow links in the BYO Infra section to read further.

#### Is there a relevant Issue open for this?
n/a

#### Who would you like to review this?
cc: @redhat-cop/casl

### CODE REVIEW

**Comment by bagnaram** [comment]
I like the use-case of breaking apart the documentation for the various infrastructures.

**Comment by etsauer** [comment]
@logandonley @oybed did either of you get a chance to test this?

File: README.md [hunk 1]
```diff
@@ -3,24 +3,25 @@
...
-The CASL Ansible tools provide everything needed to automatically provision an OpenShift cluster from scratch. Visit the provisioning guide relevant to you to get started.
+Additionally, we have several fully automated provisioners through which you can stand up an OpenShift cluster.
 
 * [Provisioning an OpenShift Cluster on OpenStack](./docs/PROVISIONING_OPENSTACK.md)
 * [Provisioning an OpenShift Cluster on AWS](./docs/PROVISIONING_AWS.md)
 * [Provisioning an OpenShift Cluster on GCP](./docs/PROVISIONING_GCP.md)
 
+### Automation of Cluster Installation on Your Own (BYO) Infrastructure
+
+Additionally, for those who are using infrastructure that is either not yet fully automated, or not yet supported through a CASL provider, we provide a [Bring Your Own Infrastructure](./docs/BYO_INFRASTRUCTURE.adoc) guide to using CASL.
```
Comment (oybed) [inline README.md]: A bit "too much repeat" starting with the same word in this section as the previous one  - maybe just remove the beginning `Additionally`.

File: docs/BYO_INFRASTRUCTURE.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+== Prepping the Inventory
+
+While the link:../inventory/sample.byo.example.com.d/[sample.byo.example.com] sample inventory provides most of the variables you need to get your cluster prepped and installed, there are a few variables in the `sample.byo.example.com.d/inventory/group_vars/all.yml` file that need to be filled in to match your environment.
+
+- The `ansible_user: <your ssh key user>` variable must match the user you will use to log into the cluster machines
+- You must fill in the `rhsm_` variables to match your Red Hat Subscription Manager account. You can do this one of two ways:
```
Comment (oybed) [inline docs/BYO_INFRASTRUCTURE.adoc]: May want to mention that `rhsm_register: True` has to be set to successfully register the hosts.

File: docs/BYO_INFRASTRUCTURE.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
++
+  - Provide a Satellite server and Activation Key:
++
+----
+rhsm_server_hostname: 'sat-6.example.com'
+rhsm_org_id: 'CASL_ORG'
+rhsm_activationkey: 'casl-latest'
+----
+
+== Running the CASL Tools on my BYO infrastructure
```
Comment (oybed) [inline docs/BYO_INFRASTRUCTURE.adoc]: Let's add `provisioning` - i.e.: "CASL Provisioning Tools"

File: docs/BYO_INFRASTRUCTURE.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Once the infrastructure has been manually provisioned and your inventory is prepped, you should be ready to run the CASL tools.
```
Comment (oybed) [inline docs/BYO_INFRASTRUCTURE.adoc]: add "provisioning" (as above)

File: inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+---
+
+deployment_type: openshift-enterprise
+openshift_deployment_type: openshift-enterprise
+openshift_master_cluster_method: native
+openshift_release: v3.7
```
Comment (oybed) [inline inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml]: Let's make it `v3.9`

File: inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+# openshift_master_cluster_public_hostname must be something like:
+#   <prefix>.{{ env_id }}.{{ dns_domain }}
+openshift_master_cluster_public_hostname: master.{{ env_id }}.{{ dns_domain }}
```
Comment (oybed) [inline inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml]: the other sample inventories uses `console` - should we do the same here?

File: inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+   'login': 'true'
+   'challenge': 'true'
+   'kind': 'HTPasswdPasswordIdentityProvider'
+   'filename': '/etc/origin/master/htpasswd'
+
+#this will create an admin/admin user
+openshift_master_htpasswd_users:
+  admin: $apr1$7aiANAYb$TOUYVUqnBqBlD5AQEIMYw1
+
+openshift_hosted_router_selector: 'region=infra'
```
Comment (oybed) [inline inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml]: These aren't really needed to be set - any particular reason to include them? (if so, we should include them for the other sample inventories as well). Same for the registry below.

File: inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+osm_default_node_selector: 'region=primary'
+
+openshift_docker_options: "--log-driver=json-file --log-opt max-size=50m --log-opt max-file=100"
+
+os_sdn_network_plugin_name: 'redhat/openshift-ovs-networkpolicy'
+#os_sdn_network_plugin_name: 'redhat/openshift-ovs-multitenant'
+os_firewall_use_firewalld: true
+osm_cluster_network_cidr: 10.1.0.0/16
+
+openshift_enable_service_catalog: false
```
Comment (oybed) [inline inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml]: Should we just leave the "disabling" of the service catalog out?

**Comment by oybed** [comment]
One other thought - with the latest doc changes, it really seems that CASL == "provisioning of OCP clusters" while in reality we are trying to make it be more than just that. Let's discuss.

File: docs/BYO_INFRASTRUCTURE.adoc [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+This guide walks through the process of using the tools in this repository to automate the stand up of OpenShift clusters on Infrastructure where there is currently no automated provisioner. This process can easily be used in on-premise installs where the infrastructure is not currently API driven and has yet to be automated in a way that's compatible with CASL.
+
+== What Infrastructure do I need before I run CASL?
+
+If you read through the link:PROVISIONING_ARCH.md[CASL Provisioning Architecture documentation], you'll notice the following description of our _Provisioning Phase_:
+
+_Provision Infrastructure_ creates the servers/instances and any other supporting infrastructure required (network, DNS, load balancers, etc.). In general this phase
+  - Begins with having credentials with proper access to an Infrastructure provider.
+  - Ends with servers provisioned and running with an OS installed and SSH keys (with root or sudo) pushed.
```
Comment (oybed) [inline docs/BYO_INFRASTRUCTURE.adoc]: May want to include some DNS info here + proper cert handling (especially when LBs are in-use)

**Comment by etsauer** [comment]
@oybed addressed your comments. I chose to leave certificates out, as they are optional and I don't want to create too high of a barrier to start using the tools. Let me know what you think

**Pull request merged**